lib/auth.js

```diff
@@ -124,15 +191,16 @@ Auth.prototype.digest = function (method, path, authHeader) {
   return authHeader
 }
 
-Auth.prototype.onRequest = function (user, pass, sendImmediately, bearer) {
+Auth.prototype.onRequest = function (user, pass, sendImmediately, bearer, negotiate) {
   var self = this
     , request = self.request
-
   var authHeader
-  if (bearer === undefined && user === undefined) {
+  if (bearer === undefined && user === undefined && negotiate === undefined) {
     self.request.emit('error', new Error('no auth mechanism defined'))
   } else if (bearer !== undefined) {
     authHeader = self.bearer(bearer, sendImmediately)
+  } else if (negotiate && kerberos !== undefined) {
+    self.negotiate(request, undefined, sendImmediately)
```

Comment:
This function will never have the SPNEGO challenge data (its returned by the 401 response), hence we always initialize the SPNEGO data with `undefined` here.  This will only be used if `sendImmediately` is set to true.